pystac/item.py

```diff
@@ -435,21 +435,21 @@ def from_dict(
         if preserve_dict:
             d = deepcopy(d)
 
-        id = d.pop("id")
-        geometry = d.pop("geometry")
-        properties = d.pop("properties")
+        id = d.pop("id", None)
+        geometry = d.pop("geometry", None)
+        properties = d.pop("properties", {})
         bbox = d.pop("bbox", None)
-        stac_extensions = d.get("stac_extensions")
+        stac_extensions = d.get("stac_extensions", None)
         collection_id = d.pop("collection", None)
 
         datetime = properties.get("datetime")
         if datetime is not None:
             datetime = str_to_datetime(datetime)
-        links = d.pop("links")
-        assets = d.pop("assets")
+        links = d.pop("links", [])
+        assets = d.pop("assets", {})
 
-        d.pop("type")
-        d.pop("stac_version")
+        d.pop("type", None)
+        d.pop("stac_version", None)
```

Comment:
I _think_ we're assuming that these are populated by `migrate_to_latest` on L428 -- can you either
1. add a comment to that effect,
2. add an assertion (or other error) just to make _sure_ they're populated, or
3. correct me?

Thanks!

Reply:
Oh! I just added these kind of blindly. I think that you are right that type and version _have_ to be provided otherwise you don't make it past `cls.matches_object_type`. But I also don't see the harm in adding None here since we are just dropping them on the floor anyway.

Reply:
I guess part of the point of this exercise is to allow `Item.__init__` to do its sort-of-validation (e.g. checking for `datetime` OR `start_datetime` OR `end_datetime`, etc), and by switching from `KeyError` to silently dropping, we lose a bit of validation IF (somehow) a dict made it through without a `type`.

Again, I don't think it can make it through, but I often like to at least `assert` these sorts of things to make it clear to my future self, who might not remember that `matches_object_type` adds/catches those.

Reply:
I added an error raise to `matches_object_type` for the case where the dict is invalid. Let me know if that feels weird. I really wanted to add the context of what field is missing that makes the dict invalid as an `Item`